photon-core/src/main/java/org/photonvision/vision/pipe/impl/Calibrate3dPoseGuidance/Distortion.java

```diff
@@ -0,0 +1,393 @@
+     * @param dpts: distorted points
+     * @param mask: mask for ignoring locations
+     * @param lower: find location with minimal distortion instead
+     * @param thres: distortion strength to use as threshold [%]
+     * @return
+     */
+    static Rect loc_from_dist(Mat pts, Mat dpts, Mat mask, boolean lower, double thres) // force specifying all parameters
+    {
+        // logger.debug("method entered  . . . . . . . . . . . . . . . . . . . . . . . .");
+        // logger.debug("pts " + pts);
+        // logger.debug("dpts " + dpts);
+        // logger.debug("mask " + mask);
+        // logger.debug("lower " + lower);
+        // logger.debug("thres " + thres);
+        Mat diffpts = new Mat();
+        Core.subtract(pts, dpts, diffpts);
+        // logger.debug("diffpts " + diffpts);
+
+        Mat normMat = new Mat(pts.rows(), pts.cols(), CvType.CV_32FC1);
+        // logger.debug("normMat empty " + normMat);
+
+        for (int row = 0; row < pts.rows(); row++)
+        for (int col = 0; col < pts.cols(); col++)
+        {
+            float[] point = new float[2]; // get the 2 channels of data x in 0 and y in 1
+            diffpts.get(row, col, point);
+            float norm = (float)Math.sqrt(Math.pow(point[0], 2) + Math.pow(point[1], 2)); // L2 norm (Frobenious)
+            normMat.put(row, col, norm);
+        }
+        // logger.debug("normMat filled " + normMat);
+
+        normMat = normMat.reshape(0,mask.rows())/*.t()*/;
+        // logger.debug("normMat reshaped " + normMat);
+
+        Mat diff = new Mat();
+        Core.normalize(normMat, diff, 0, 255, Core.NORM_MINMAX, CvType.CV_8U);
+        // logger.debug("diff " + diff.dump());
+        // logger.debug("normMat normalized=diff " + diff);
+
+        Rect bounds = null;
+
+        Mat thres_img = new Mat();
+        while ((bounds == null) && (thres >= 0.) && (thres <= 1.) )
+        {
+            if (lower)
+            {
+                thres += 0.05;
+                Imgproc.threshold(diff, thres_img, thres * 255., 255., Imgproc.THRESH_BINARY_INV);
+            }
+            else
+            {
+                thres -= 0.05;
+                Imgproc.threshold(diff, thres_img, thres * 255., 255., Imgproc.THRESH_BINARY);
+            }
+            // logger.debug("thres_img " + thres_img /*+ "\n" + brief(thres_img.dump())*/);
+
+            bounds = get_bounds(thres_img, mask);
+
+            if (bounds == null)
+            {
+                continue;
+            }
+
+            if (bounds.width*bounds.height == 0) // ensure area is not 0
+            {
+                bounds = null;                
+            }
+        }
+
+        normMat.release();
+        diff.release();
+        diffpts.release();
+
+        // logger.debug("bounds " + (bounds == null ? "is null" : bounds));
+
+        return bounds;
+    }
+/*-------------------------------------------------------------------------------------------------*/
```

Comment:
Check out the other java files for the format used on method headers. Not that we use doxygen a ton, but it does help some code editors if the headers are in a standard format.

Reply:
A year ago when Matt suggested this project, "automated" calibrating pose selection might have been a somewhat useful addition to PV. Now it doesn't appear to be so.

I hesitate to kill this project only because of my nearly 900 hours to do this Python to Java conversion. I had most of the code to integrate calibration pose with PV. Nobody specified how the selection of the feature should appear on the PV user interface. And I didn't know much about the coding to do the final couple of hours to pass the user check box (or whatever was needed) and I didn't want to take a lot of time learning what I'd never again use. Nobody else stepped in to help finish up my project as they were all busy and had their own more interesting work to do.

Now months later I see another person or team added to PV somewhat conflicting calibration board work without thought to using or merging with what I had started. My project is 99 commits behind PV and the merge conflicts have become very significant. This pose calibration project went from nearly done to needs much work that nobody wants to do. Matt said it was my choice if I wanted to finish the project or not. That's a tepid specification of the usefulness of this project and I take it that the project might not even be used even if perfectly completed.

I suppose I should close the pull request as I've gone as far as I reasonably can but came up short of helping PV.

Reply:
> Now it doesn't appear to be so.

I want to be clear, this is objectively false. The functionality of giving hints to users to streamline the calibration process has been repeatedly requested in user feedback, discussed on discord, and mentioned in team meetings.

It's unfortunate this came up for review in January - it has enough risk that it wasn't likely to get merged during build season, and most developers didn't have the time to jump in and do something specific with it, due to the season being active.

However, this isn't an acceptable response to PR review comments in a volunteer project. A simple "Hey, I don't think I have the time to work this anymore" would have sufficed.

I cannot figure out where Cameron has put the implementation discussed on discord in April. If you believe your approach is further along and more viable than Cameron's, let's start addressing the review comments to get this over the line. 

Otherwise, yes, please close out this PR so others do not spend time reviewing it. The changes will always be available for reference when someone else starts to implement the functionality.